src/bindings/python/src/openvino/_ov_api.py

```diff
@@ -600,10 +604,8 @@ def compile_model(
             )
         else:
             if device_name is None:
-                return CompiledModel(
-                    super().compile_model(model, weights, {} if config is None else config),
-                    weights=weights,
-                )
+                raise ValueError("device_name must be provided when weights are specified.")
```

Comment:
please, remove this `if device_name is None:` and `raise ValueError`.
There should be thrown the default error by pybind11 with the list of available arguments for `compile_model` method